src/bandersnatch/master.py

```diff
@@ -41,13 +41,15 @@ def __init__(
         timeout: float = 10.0,
         global_timeout: float | None = FIVE_HOURS_FLOAT,
         proxy: str | None = None,
+        allow_non_https: bool | None = False,
```

Comment:
```suggestion
        allow_non_https: bool = False,
```
I don't see the advantage of None, unless we want to log saying it's not set in config ... but we don't.